src/users/users.controller.ts

```diff
@@ -0,0 +1,51 @@
+import { Controller, Post, Body, Put, Param } from '@nestjs/common';
+import { IsString } from 'class-validator';
+import { UsersService } from './users.service';
+import { ObjectID } from 'mongodb';
+import { User } from './users.model';
+
+class RegisterDto {
+    @IsString()
+    public uuid: string;
+
+    @IsString()
+    public password: string;
+}
+
+export class LoginDto {
+    @IsString()
+    public uuid: string;
+
+    @IsString()
+    public password: string;
+}
+
+export class PatchUserDto {
+    @IsString()
+    public name: string;
+    @IsString({ each: true })
+    public locations: string[];
+}
+
+@Controller('users')
+export class UsersController {
+    constructor(private users: UsersService) {}
+
+    @Post('register')
+    public async register(@Body() body: RegisterDto): Promise<User> {
+        const user = await this.users.register(body);
+        return user.serialize();
+    }
+
+    @Post('login')
+    public async login(@Body() body: LoginDto): Promise<User> {
```

Comment:
For the 1st version I think we should do both login and register together.
```
if (login == false && uuidExists = true) {
throw
}
if (login == false && uuidExists = false) {
signup
}

if (login === true) {
success
}
```
App wise it would be super simple